src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -3475,41 +3475,35 @@
       <Kind Name="SemicolonToken"/>
     </Field>
   </Node>
-  <Node Name="RoleDeclarationSyntax" Base="TypeDeclarationSyntax" SkipConvenienceFactories="true">
-    <Kind Name="RoleDeclaration" />
-    <Field Name="AttributeLists" Type="SyntaxList&lt;AttributeListSyntax&gt;" Override="true"/>
-    <Field Name="Modifiers" Type="SyntaxList&lt;SyntaxToken&gt;" Override="true"/>
-    <Field Name="Keyword" Type="SyntaxToken" Override="true">
-      <ContextualKind Name="RoleKeyword"/>
-    </Field>
-    <Field Name="Identifier" Type="SyntaxToken" Override="true">
-      <Kind Name="IdentifierToken"/>
-    </Field>
-    <Field Name="TypeParameterList" Type="TypeParameterListSyntax" Optional="true" Override="true"/>
-    <Field Name="BaseList" Type="BaseListSyntax" Optional="true" Override="true"/>
-    <Field Name="ConstraintClauses" Type="SyntaxList&lt;TypeParameterConstraintClauseSyntax&gt;" Override="true"/>
-    <Field Name="OpenBraceToken" Type="SyntaxToken" Override="true" Optional="true">
-      <Kind Name="OpenBraceToken"/>
-    </Field>
-    <Field Name="Members" Type="SyntaxList&lt;MemberDeclarationSyntax&gt;" Override="true"/>
-    <Field Name="CloseBraceToken" Type="SyntaxToken" Override="true" Optional="true">
-      <Kind Name="CloseBraceToken"/>
-    </Field>
-    <Field Name="SemicolonToken" Type="SyntaxToken" Optional="true" Override="true">
-      <Kind Name="SemicolonToken"/>
+  <Node Name="ForTypeSyntax" Base="CSharpSyntaxNode">
+    <TypeComment>
+      <summary>Class representing a 'for' token followed by a type.</summary>
+    </TypeComment>
+    <Kind Name="ForType"/>
+    <Field Name="ForKeyword" Type="SyntaxToken">
+      <Kind Name="ForKeyword"/>
     </Field>
+    <Field Name="UnderlyingType" Type="TypeSyntax"/>
   </Node>
   <Node Name="ExtensionDeclarationSyntax" Base="TypeDeclarationSyntax" SkipConvenienceFactories="true">
     <Kind Name="ExtensionDeclaration" />
     <Field Name="AttributeLists" Type="SyntaxList&lt;AttributeListSyntax&gt;" Override="true"/>
     <Field Name="Modifiers" Type="SyntaxList&lt;SyntaxToken&gt;" Override="true"/>
+    <Field Name="ImplicitOrExplicitKeyword" Type="SyntaxToken">
+      <PropertyComment>
+        <summary>Gets the "implicit" or "explicit" token.</summary>
+      </PropertyComment>
+      <Kind Name="ImplicitKeyword"/>
+      <Kind Name="ExplicitKeyword"/>
+    </Field>
     <Field Name="Keyword" Type="SyntaxToken" Override="true">
       <ContextualKind Name="ExtensionKeyword"/>
     </Field>
     <Field Name="Identifier" Type="SyntaxToken" Override="true">
       <Kind Name="IdentifierToken"/>
     </Field>
     <Field Name="TypeParameterList" Type="TypeParameterListSyntax" Optional="true" Override="true"/>
+    <Field Name="ForUnderlyingType" Type="ForTypeSyntax" Optional="true"/>
```

Comment:
not sure how i feel about a whole node for this.  but i can live with it :)

Reply:
I would agree, this doesn't seem like it needsa node. Feels like it should be a `For` token and an underlying type node.

Reply:
We have a node for name-equals. I think the same model applies here, since the pair (for+type) is optional, but the individual parts are not.

Reply:
From offline discussion, Fred was okay with the current structure to handle optionality, but I captured a bullet in test plan to highlight/discuss this choice in API review. Thanks